src/adapter/src/catalog/transact.rs

```diff
@@ -482,9 +557,14 @@ impl Catalog {
                     storage_collections_to_drop,
                 )
                 .await?;
-            }
```

Comment:
If storage_controller is None, this is supposed to do nothing? Can this get a comment for why it's right to do that?

Reply:
Sorry, I think I'm having trouble understanding exactly what you're asking, specifically what "this is supposed to do nothing" means. Are you talking about the existing code?

```
            if let Some(c) = storage_controller {
                c.prepare_state(
                    tx,
                    storage_collections_to_create,
                    storage_collections_to_drop,
                )
                .await?;
            }
```

Or the code I added specifically?

Reply:
Ah ok, I just pulled the new block outside of the `if let Some` to match the old behavior and added a comment about this section. `storage_controller` should only be `None` for tests, which makes this whole section a bit difficult to reason about. IMO, it should not be an option and instead mocked in test, but that's probably not a change I'll make in this PR.